components/camel-bindy/src/main/java/org/apache/camel/dataformat/bindy/csv/BindyCsvDataFormat.java

```diff
@@ -138,58 +144,74 @@ public Object unmarshal(Exchange exchange, InputStream inputStream) throws Excep
     
             in = new InputStreamReader(inputStream, IOHelper.getCharsetName(exchange));
-    
-            // Scanner is used to read big file
-            scanner = new Scanner(in);
-    
+
             // Retrieve the separator defined to split the record
             String separator = factory.getSeparator();
             String quote = factory.getQuote();
             ObjectHelper.notNull(separator, "The separator has not been defined in the annotation @CsvRecord or not instantiated during initModel.");
-    
-            int count = 0;
-            
-            // If the first line of the CSV file contains columns name, then we
-            // skip this line
-            if (factory.getSkipFirstLine()) {
-                // Check if scanner is empty
-                if (scanner.hasNextLine()) {
-                    scanner.nextLine();
+            AtomicInteger count = new AtomicInteger(0);
+
+            // Use a Stream to stream a file across.
+            try (Stream<String> lines = new BufferedReader(in).lines()) {
+                int linesToSkip = 0;
+
+                // If the first line of the CSV file contains columns name, then we
+                // skip this line
+                if (factory.getSkipFirstLine()) {
+                    linesToSkip = 1;
                 }
-            }
-    
-            while (scanner.hasNextLine()) {
-    
-                // Read the line
-                String line = scanner.nextLine().trim();
-    
-                if (ObjectHelper.isEmpty(line)) {
-                    // skip if line is empty
-                    continue;
+
+                // Consume the lines in the file via a consumer method, passing in state as necessary.
+                // If the internals of the consumer fail, we unrap the checked exception upstream.
+                try {
+                    lines.skip(linesToSkip)
+                            .forEachOrdered(consumeFile(factory, models, separator, quote, count));
+                } catch (WrappedException e) {
+                    throw e.getWrappedException();
                 }
-    
+
+                // BigIntegerFormatFactory if models list is empty or not
+                // If this is the case (correspond to an empty stream, ...)
+                if (models.size() == 0) {
+                    throw new java.lang.IllegalArgumentException("No records have been defined in the CSV");
+                } else {
+                    return extractUnmarshalResult(models);
+                }
+            }
+        } finally {
+            if (in != null) {
+                IOHelper.close(in, "in", LOG);
+            }
+        }
+
+    }
+
+    private Consumer<String> consumeFile(BindyCsvFactory factory, List<Map<String, Object>> models,
+                                         String separator, String quote, AtomicInteger count) {
+        return line -> {
+            try {
+                // Trim the line coming in to remove any trailing whitespace
```

Comment:
Long time no use of bindy for myself, is not there an option for bindy fields to trim or not already?

Reply:
This was [existing behavior](https://github.com/apache/camel/blob/master/components/camel-bindy/src/main/java/org/apache/camel/dataformat/bindy/csv/BindyCsvDataFormat.java#L174).  I removed it in error while working on this PR and found tests broke with it removed.